activesupport/lib/active_support/key_generator.rb

```diff
@@ -8,6 +8,9 @@ module ActiveSupport
   # key in multiple incompatible contexts.
   class KeyGenerator
     def initialize(secret, options = {})
+      unless secret.kind_of?(String)
+        raise ArgumentError, "Secret must be a type of String"
```

Comment:
Maybe we should check it were we assign with a proper message pointing to the secrets configuration. I think if we raise here it will not be user friendly.